### DISCUSSION TITLE
Record and Replay keyed topics

### DISCUSSION THREAD

**Discussion by manuelValch** [eProsima/DDS-Record-Replay#170] [General]
Hello, 

Working on some project with fastDDS and keyed topics, and I tried to use the recorder on it.

Though recorded data does not seems to have all data (with all keys), if anyone has done it revently would appreciate hints 

I am still using v0.3.0, I know some work has been done on Key handling, shall I update my tool version ?

Regards

**Comment by juanlofer-eprosima** [comment]
Hi @manuelValch ,

Could you please be more specific about the problem? We frequently test with keyed topics with no issues. Could you provide the idl with your type description? It would also help to know which is the source of the data being recorded (so we know which middleware version is used). And yes, there has been a deep refactor in the X-types module so we encourage to update to the latest release if possible.

**Comment by manuelValch** [comment]
Hello @juanlofer-eprosima 

So version of fast dds I am using is 2.13.1 (a quite old one), I know that there were some changes performed on key topic between 2.13 and 2.14, fastdds recorder version is 0.3.0.
Here is the idl:
```
struct message
{
    @Key unsigned long emitterId; /// key name of the msg transmitter
    unsigned long index; /// msg index
    string messageStr; /// message string
};
```

I have tested it without anything set in yaml config file, seems to be working fine (I have all key registered)

Sorry for the disturbance,though I will definitly concider updating on recent version of fastdds (e.g 3.x.x) with refactor of X type.

Thank you
